### DISCUSSION TITLE
Protected api routes also works with data fetching in server component ?

### DISCUSSION THREAD

**Discussion by GomaGoma676** [nextauthjs/next-auth#6103] [Help] [answered]
Hi,

Currently, I'm playing around following example repo of next.js v13 app dir, especially protected api routes.
I already confirmed that protected api routes with unstable_getServerSession working fine if I data fetch from client component but not working with data fetching from server component (always data fetch error even authenticated). is it technically not possible ?

https://github.com/nextauthjs/next-auth/blob/main/apps/dev/pages/api/examples/protected.js

**Comment by GomaGoma676** [comment, marked as answer]
After further play around, here is one solution how to access to protected api route endpoint from server component.

page.tsx
```
import { cookies } from 'next/headers'

async function fetchNotes(token: string | undefined) {
  const res = await fetch(
    `your-api-url`,
    {
      headers: {
        cookie: `next-auth.session-token=${token}`,
      },
    }
  )
  if (!res.ok) {
    throw new Error('Failed to fetch data in server')
  }
  const notes = await res.json()
  return notes
}
export default async function Page() {
  const nextCookies = cookies()
  const token = nextCookies.get('next-auth.session-token')
  const notes = await fetchNotes(token?.value)

  return (
    <>
    </>
  )
}
```

**Comment by laurent512** [comment]
I struggled on that point as well ! Hope a simpler solution will be brought later on
